testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -0,0 +1,511 @@
+    cookie, session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=username)
+    auth_session = cookie, session
+    sync_cookie = "{}={}".format(cookie, session)
+    session_header = {"Cookie": sync_cookie}
+
+    # Replicating docs to SG so that later we can create conflict
+    # Adding Listener for replicator
+    replicator = Replication(base_url)
+    repl_config = replicator.configure(source_db=cbl_db,
+                                       target_url=sg_blip_url,
+                                       continuous=False,
+                                       headers=session_header,
+                                       replication_type="push")
+    repl = replicator.create(repl_config)
+    repl_change_listener = replicator.addReplicatorEventChangeListener(repl)
+
+    # 2. Starting Replication and waiting for it finish
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # 3. Adding conflicts for docs in SG
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)
+    sg_docs = sg_docs["rows"]
+    sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs, number_updates=1, auth=auth_session)
+
+    # Adding conflict for docs in CBL
+    db.update_bulk_docs(database=cbl_db, number_of_updates=1,
+                        doc_ids=cbl_docs)
+
+    # 4. Starting one-shot replication again to create error based on conflict
+    repl_conflict_config = replicator.configure(source_db=cbl_db,
+                                                target_url=sg_blip_url,
+                                                continuous=False,
+                                                headers=session_header,
+                                                replication_type="push")
+    repl_conflict = replicator.create(repl_conflict_config)
+    repl_error_change_listener = replicator.addReplicatorEventChangeListener(repl_conflict)
+    replicator.start(repl_conflict)
+    replicator.wait_until_replicator_idle(repl_conflict)
+
+    # 5. verifying the error in the events collected
+    doc_error_repl_event_changes = replicator.getReplicatorEventChanges(repl_error_change_listener)
+    replicator.removeReplicatorEventListener(repl_conflict, repl_error_change_listener)
+    replicator.stop(repl_conflict)
+    event_dict = get_event_changes(doc_error_repl_event_changes)
+    assert len(event_dict) != 0, "No Event captured. Check the logs for detailed info"
+    for doc_id in event_dict:
+        assert '10409' in event_dict[doc_id]["error_code"], "Conflict error didn't happen. Error Code: {}".format(
+            event_dict[doc_id]["error_code"])
+
+
+# @pytest.mark.sanity
+# @pytest.mark.listener
+# @pytest.mark.replication
+# @pytest.mark.parametrize("num_of_docs", [
+#     1000,
+# ])
+# def test_pull_replication_error_event(params_from_base_test_setup, num_of_docs):
```

Comment:
is this functionality not yet implemented , that is why it is commented?

Reply:
Yes

Reply:
can we remove this test then?